File: Common/Entity/src/main/java/it/ey/dto/Sezione1DTO.java
package it.ey.dto;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class Sezione1DTO extends SezioneBaseDTO{
    private Long id;
    private Long idPiao;
    private String statoSezione;

    private String quadroNormativo;
    private String strutturaProgrammatica;
    private String cronoprogramma;
    private String missione;
    private List<AreaOrganizzativaDTO> areeOrganizzative;
    private List<PrioritaPoliticaDTO> prioritaPolitiche;
    private List<PrincipioGuidaDTO> principiGuida;
    private List<OrganoPoliticoDTO> organiPolitici;
    private List<IntegrationTeamDTO> integrationTeams;
    private AnagraficaDTO anagrafica;
    private List<StakeHolderDTO> stakeHolders;



}

File: PIAO_FE/src/app/shared/models/classes/sezione-1-dto.ts
import { AllegatoDTO } from './allegato-dto';
import { AreaOrganizzativaDTO } from './area-organizzativa-dto';
import { CampiTecniciDTO } from './campi-tecnici-dto';
import { IntegrationTeamDTO } from './integration-team-dto';
import { OrganoPoliticoDTO } from './organo-politico-dto';
import { PrincipioGuidaDTO } from './principio-guida-dto';
import { PrioritaPoliticaDTO } from './priorita-politica-dto';
import { SocialDTO } from './social-dto';
import { StakeHolderDTO } from './stakeholder-dto';
import { UlterioriInfoDTO } from './ulteriori-info-dto';

export class Sezione1DTO extends CampiTecniciDTO {
  id?: number;
  idPiao?: number;
  statoSezione?: string;
  quadroNormativo?: string;
  strutturaProgrammatica?: string;
  cronoprogramma?: string;
  missione?: string;
  areeOrganizzative?: AreaOrganizzativaDTO[];
  prioritaPolitiche?: PrioritaPoliticaDTO[];
  principiGuida?: PrincipioGuidaDTO[];
  organiPolitici?: OrganoPoliticoDTO[];
  integrationTeams?: IntegrationTeamDTO[];
  ulterioriInfoDTO?: UlterioriInfoDTO;
  stakeHolders?: StakeHolderDTO[];
  allegati?: AllegatoDTO[];
  social?: SocialDTO;
}
